modules/internal/SharedObject.chpl

```diff
@@ -416,6 +420,9 @@ module SharedObject {
      managed by ``rhs``, and ``rhs`` will refer to `nil`.
    */
   proc =(ref lhs:_shared, in rhs:owned) {
+    if isNonNilableClass(lhs) && isNilableClass(rhs) then
+      compilerError("cannot assign to a non-nilable shared variable from a nilable class instance");
```

Comment:
I would expect

"cannot assign to a non-nilable shared variable from a nilable owned"